lib/truffle/truffle/patching.rb

```diff
@@ -13,11 +17,16 @@ module Kernel
   alias_method :require_without_truffle_patching, :require
 
   def require(path)
+    if Truffle::Patching::TRUFFLE_BEFORE_PATCHES[path]
```

Comment:
Yes, otherwise every require 'openssl' will reload the patch file.
One easy fix is to use `require_without_truffle_patching` for loading the patch instead of `load`.
It does mean an extra Hash lookup for every require though.